mmd/compositeconn.go

```diff
@@ -161,21 +165,43 @@ func (c *CompositeConn) close() (err error) {
 	return
 }
 
-func (c *CompositeConn) getOrCreateConnection(service string) (*ConnImpl, error) {
-	log.Println("get or create connection " + service)
+func (c *CompositeConn) onDisconnect(connCounter int32) {
```

Comment:
per offline, remove `er` from here, and maybe rename
```suggestion
func (c *CompositeConn) onDisconnect(connCount int32) {
```